### PR TITLE
api call for category and some code refactor

### PR DESCRIPTION
**Pull request by ShreyashMahajan**
In this PR, an API call for the category has been done. 
Some code has been refactored. 
please review the pr and provide feedback.

### CODE REVIEW

File: src/context/filterContext/filterContext.jsx [hunk 1]
```diff
@@ -16,11 +16,17 @@ const FilterProvider = ({ children }) => {
...
+    const dataBySearch = filterBySearch(
+        filterDataByCategory(
+            filterByPrice(
+                filterByRating(
+                    sortDataByRange(
+                        productList, state.sortbyrange),
+                    state.sortByRating), state.priceValue),
+            state.category.isBike, state.category.isScooter),
+        state.searchItem);
+
```
Comment (sushil-mod) [inline src/context/filterContext/filterContext.jsx:29]: i liked this f(g(x)) for filtering data

**Review by sushil-mod — APPROVED** [review]
category Filter from home page working fine , good code

File: src/context/filterContext/filterContext.jsx [hunk 2]
```diff
@@ -16,11 +16,17 @@ const FilterProvider = ({ children }) => {
     const { productList } = useProduct();
     const [state, dispatch] = useReducer(filterReducer, { sortbyrange: "", sortBySearch: "", sortByRating: "", priceValue: 160000, category: { isBike: false, isScooter: false } });
 
-    const sortedData = sortDataByRange(productList, state.sortbyrange);
-    const filteredData = filterByRating(sortedData, state.sortByRating);
-    const filterDataByPrice = filterByPrice(filteredData, state.priceValue);
-    const dataByCategory = filterDataByCategory(filterDataByPrice, state.category.isBike, state.category.isScooter);
-    const dataBySearch = filterBySearch(dataByCategory, state.searchItem);
+
+    const dataBySearch = filterBySearch(
```
Comment (ankit9015) [inline src/context/filterContext/filterContext.jsx:20]: good use of composite functions

**Review by ankit9015 — APPROVED** [review]
PR is good to merge. 
1.  I like how you are using images from public 
2.  Composite function for filter products
3.  Code is well written and good folder structure is used

File: src/context/filterContext/filterContext.jsx (hunk 1 above)
Comment (ShreyashMahajan) [reply src/context/filterContext/filterContext.jsx:29]: thank you

File: src/context/filterContext/filterContext.jsx (hunk 2 above)
Comment (ShreyashMahajan) [reply src/context/filterContext/filterContext.jsx:20]: thank you

**Pull request merged**